libraries/common/stream.effekt

```diff
@@ -0,0 +1,234 @@
+module stream
+
+import bytes
+
+interface Emit[A] {
+  def emit(value: A): Unit
+}
+
+def for[A, R] { stream: () => R / Emit[A] } { action: A => Unit }: R =
+  try {
+    stream()
+  } with Emit[A] {
+    def emit(value) = resume(action(value))
+  }
+
+// TODO what if the number is negative? undefined behavior?
```

Comment:
I'd want the program to either panic or act as if it is 0. (This is not something where I'd expect forever, and this is a place where I could see myself accidentally passing a negative number).

Reply:
Yes, please check and panic

Reply:
If we panic, we have to call it `unsafeRepeat`. I'm doing nothing for now. I wish we had `Nat`.